### PR TITLE
bring sigs to the front of the website

### LINKED ISSUE #66: bring sigs to the front of the website

### PR DESCRIPTION
**Pull request by JMaio**
[live testing link](https://jmaio.github.io/website2017/)
resolves #66 
- adds reusable element `sigs-carousel.html` to be used on front page
  - dependencies: `jQuery`, `owl-carousel`

also alters:
- front page height (shrink compsoc logo and padding to mostly fit onscreen)
- how SIGs are arranged in the website, being given dynamic urls to subpages based on path
- SIGs data file has new attributes:
  - `url` - if present, links to external, otherwise, `slug` is used to link to subpage
  - `image` - boolean to indicate whether image exists; if true, uses `slug` to dynamically locate image in `/static/img/sigs/{slug}.png`
  - `colour` - set background colour to contrast image, can also be `none`
  - `radius` - used for css border-radius property

### CODE REVIEW

File: _includes/sigs-carousel.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<div class="owl-carousel owl-drag owl-theme">
+  <!-- <div id="front-sigs" class="slidee d-flex flex-wrap justify-content-center justify-content-sm-start mb-2"> -->
```
Comment (neanias) [inline _includes/sigs-carousel.html]: We should remove commented out lines.

File: _includes/sigs-carousel.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        smartSpeed: 1500,
+        // fluidSpeed: 400,
+        dragEndSpeed: 200,
+      },
+      dots: true,
+      responsiveRefreshRate: 100,
+    },
+  });
+  owl.on('mousewheel', '.owl-stage', function (e) {
+    if (e.deltaY>0) {
```
Comment (neanias) [inline _includes/sigs-carousel.html]: Space around this operator?

**Comment by qaisjp** [comment]
Had a quick look from your testing page, it looks beautiful. Well done. I'll give it a full review soon.

File: _pages/sigs.md [hunk 3]
```diff
@@ -27,7 +10,7 @@ CompSoc offers SIGs promotion, some money, support, and a place on the committee
 
 ### Our SIGs
 <div class="d-flex flex-wrap justify-content-center justify-content-sm-start mb-2">
-    <div class="sigs-item" style="background-color: #292b2c">
+    <!-- <div class="sigs-item" style="background-color: #292b2c">
```
Comment (neanias) [inline _pages/sigs.md]: Commented out

**Comment by neanias** [comment]
Perhaps use the same code as we use for `[citation needed]` for the SIGs definition?

**Comment by qaisjp** [comment]
Happy to approve this once @neanias' changes are applied. Really well done :) It's beautiful!

File: _pages/sigs.md (hunk 3 above)
Comment (neanias) [reply _pages/sigs.md]: Here, still

File: _sass/_index-jumbotron.sass [hunk 4]
```diff
@@ -1,3 +1,7 @@
 #index-jumbotron
     /* Background pattern from Subtle Patterns */
     background-image: url($img + "pw_pattern.png")
+
+.jumbotron
```
Comment (qaisjp) [inline _sass/_index-jumbotron.sass]: Why is this applied to `.jumbotron` and not `#index-jumbotron`? Also `class="pt-0"` is a nice alternative

Comment (JMaio) [reply _sass/_index-jumbotron.sass]: no reason not to use `#index-jumbotron`, will include.
`.pt-0` - is this new in Bootstrap4? haha

Comment (qaisjp) [reply _sass/_index-jumbotron.sass]: haha yes i think so, there's a ton of new utility classes

**Pull request merged**